### PR TITLE
feat: campaign missing pieces

### PR DESCRIPTION
**Pull request by ferhatelmas**
it has some breaking changes but this API had been marked as to be changed.

### CODE REVIEW

File: src/client.ts [hunk 1]
```diff
@@ -2683,26 +2689,20 @@ export class StreamChat<StreamChatGenerics extends ExtendableGenerics = DefaultG
...
-   * listSegments - List Campaign Segments
+   * querySegments - Query Campaign Segments
    *
    *
    * @return {Segment[]} Segments
    */
-  async listSegments(options: { limit?: number; offset?: number }) {
-    const { segments } = await this.get<{ segments: Segment[] }>(this.baseURL + `/segments`, options);
-    return segments;
+  async querySegments(filters: SegmentFilters, options: CampaignQueryOptions = {}) {
```
Comment (SiddhantAgarwal) [inline src/client.ts:2697]: `CampaignQueryOptions` could be changed to something related to segments options.

File: src/client.ts [hunk 2]
```diff
@@ -2843,6 +2840,27 @@ export class StreamChat<StreamChatGenerics extends ExtendableGenerics = DefaultG
     return await this.post<APIResponse & TestCampaignResponse>(this.baseURL + `/campaigns/${id}/test`, { users });
   }
 
+  /**
+   * queryRecipients - Query Campaign Recipient Results
+   *
+   *
+   * @return {Recipient[]} Recipients
+   */
+  async queryRecipients(filters: RecipientFilters, options: CampaignQueryOptions = {}) {
```
Comment (SiddhantAgarwal) [inline src/client.ts:2849]: `CampaignQueryOptions` same for query recipients

File: src/client.ts (hunk 1 above)
Comment (ferhatelmas) [reply src/client.ts:2697]: It's same type but we can provide alias

Comment (ferhatelmas) [reply src/client.ts:2697]: Let me add alias them so that we can diverge if we need later on

**Review by SiddhantAgarwal — APPROVED** [review]

**Pull request merged**